### PR TITLE
str_utils: add multiline BFW-970

### PR DESCRIPTION
**Pull request by espr14**

### CODE REVIEW

**Comment by espr14** [comment]
str_utils are redone nearly from scratch so using of diff tool does not make much sense.

File: src/common/str_utils.h [hunk 1]
```diff
@@ -1,43 +1,33 @@
...
+#define NBSP "\xA0"                 /// Non Breaking Space
+constexpr char CHAR_NL = '\n';      /// New Line
+constexpr char NL[2] = { CHAR_NL }; /// New Line
+constexpr char EOS = '\0';          /// end of string
 
-#define QT_HSPACE "\x1F" // ~ <UnitSeparator>
-#define QT_LF     "\n"   // ~ <LineFeed>
-#define QT_NL     QT_LF
-#define QT_HYPHEN "\x1A" // ~ <Substitute>
+#ifdef __cplusplus
```
Comment (RadekVana) [inline src/common/str_utils.h:13]: it is included in cpp files only
remove `#ifdef __cplusplus` ?

Comment (RadekVana) [reply src/common/str_utils.h:13]: and please rename headers to .hpp

File: src/common/str_utils.h [hunk 2]
```diff
@@ -1,43 +1,33 @@
...
+extern "C" int str2multiline(char *str, size_t max_size, const size_t line_width);
 
-#define EOS '\x00'
+#else
 
-enum class delimiter_t : uint8_t { NONE,
-    SPACE,
-    HYPHEN,
-    CUSTOM };
+typedef struct
```
Comment (RadekVana) [inline src/common/str_utils.h:22]: please do not use `typedef struct` in C++ code

File: src/common/str_utils.cpp [hunk 3]
```diff
@@ -1,168 +1,142 @@
...
+        return -1;
+    if (n == 0)
+        return 0;
 
-void set_hyphen_distance(int dist) {
-    hyphen_distance = dist;
-}
+    const size_t size = strlen(str);
+    if (size + n >= max_size) /// too much to add
+        return -2;
```
Comment (RadekVana) [inline src/common/str_utils.cpp:35]: define some constants for -1 and -2 return values

**Pull request merged**